### PR TITLE
WIP: Platen Heat Exchanger

### PR DESCRIPTION
**Pull request by PTA0001**
## Fixes
1. Removes direct call to Cp parameters in the  unit model

## Changes proposed in this PR:
- Updates the plate heat exchanger model to make use of the generic property framework

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (MAZamarripa) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:431]: @andrewlee94 add to mass flow to property package (inkg/s)

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (MAZamarripa) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:446]: in line comment with better description

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (MAZamarripa) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:495]: channel_diameter

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:255]: The names for these need to be more descriptive. `P` is an especially poor choice as it invites confusion with pressure.

Also, does this actually need to be a `Param`? If it is only used for building the sets below, then I don't think it is necessary and we could just use the config argument directly.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:260]: If these will be equal, is it necessary to make two Sets?

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:266]: It would be nice if these units were not hard coded and depended on the property package instead.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:300]: This looks like a very bad idea. Why is the config argument itself appearing in the Expression - this looks like it should just be a unit level `Var`.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:314]: As above - it seems like a really bad idea to build the config arguments directly into the Expressions.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:351]: What exactly is this meant to be? Specifically, what is the correct value if `j` is not a solvent? Should this return `Expression.Skip` if `j` is not a solvent?

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:427]: Hard coded units here will be a problem.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:464]: This will probably need to have unit conversions to ensure consistency (unless you ensure that the spatial dimensions are consistent with the property package).

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:525]: Cold side

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:536]: This will need unit conversions too.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:684]: Unit conversion.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:690]: What is the purpose of Pset here?

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:225]: Where is the energy balance for the system? Even with the NTU calculations, I would expect to see an proper energy balance for each control volume with a heat transfer term, a constraint that says Qhot = -Qcold, and then the NTU calculations used to determine the value of one of the Q terms.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
1. This PR will need to wait until some of the other PRs have been merged. It is expected that it will depend on the SovlentReboiler PR, but I also see some other things have slipped in.
2. There are a lot of areas where unit conversions are needed- at the moment this model presumes units on everything.
3. I do not see a proper energy balance in this model.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [reply idaes/generic_models/unit_models/plate_heat_exchanger.py:431]: Mass flow should already be a property.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:448]: Where do these hard coded constants come from? There should at a minimum be a comment here with an explanation or reference for the source of these.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (PTA0001) [reply idaes/generic_models/unit_models/plate_heat_exchanger.py:464]: @andrewlee94, this might be the source of the unit consistency(applies to other similar comments), I will take the first pass at converting this but let's look at it during our 10 am meeting today.

**Comment by PTA0001** [comment]
@andrewlee94 , @MAZamarripa , @anujad95 
I have updated the phe to capture the review comments. On the unit consistency, while trying to convert the unit for dynamic viscosity it gives an error that Pa/s not equal to Pa.s.  Hope to discuss this in our meeting today.(I commented the code out)

**Comment by andrewlee94** [comment]
@PTA0001 There is an import error in your code somewhere that is preventing the tests from running. This really should be addressed before we review the code so that we can confirm that the tests pass.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:369]: Are these really `Vars`? These look like coefficients for a correlation, and thus I question whether these should be varied during optimization of the unit.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:387]: As above.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:490]: Descriptive names.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [inline idaes/generic_models/unit_models/plate_heat_exchanger.py:492]: Again, I would like to know where these hard-coded constants are coming from. A comment may suffice.

File: idaes/generic_models/unit_models/plate_heat_exchanger.py
Comment (andrewlee94) [reply idaes/generic_models/unit_models/plate_heat_exchanger.py:536]: @PTA0001 a quick check of the property packages shows that the units were accidentally set to Pa/s instead of Pa.s (line 450 in https://github.com/IDAES/idaes-pse/blob/main/idaes/power_generation/carbon_capture/mea_solvent_system/properties/MEA_solvent.py).

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
I had a look through the changes and added a few more comments. Overall, it is coming along but I still see a number of things that need to be fixed to meet a minimum standard:

1. The hot and cold side control volumes need to have energy balances added with a heat transfer term, and there needs to be a constraint that says `Q_hot == -Q_cold`. The NTU method can then be used to define the value of `Q`.
2. Every component should have a descriptive name, not just the ones I explicitly point out.
3. There are a number of hard-coded constants that appear in the model which should be explained in comments at the least.

I also have a theoretical question about the way the model is structured. You are using the overall inlet and outlet conditions to get the `cp` values and using them across the entire unit. However, internally the model is broken down into sub-exchangers (each plate pass I believe). This would seem to imply that you really should be calculating `cp` at the inlet and outlet of each sub-exchanger; alternatively you could just apply the NTU method across the exchanger as a whole with some correlation for `eps` that accounts for the multiple passes.

**Comment by andrewlee94** [comment]
Replaced by #590.

**Pull request closed without merging**
